src/platform_impl/linux/wayland/event_loop.rs

```diff
@@ -714,8 +714,43 @@ impl<T> EventLoop<T> {
         window_target.store.lock().unwrap().for_each(|window| {
             let window_id =
                 crate::window::WindowId(crate::platform_impl::WindowId::Wayland(window.wid));
-            if let Some(frame) = window.frame {
-                if let Some((w, h)) = window.newsize {
+
+            // Update window logical .size field (for callbacks using .inner_size)
+            let (old_size, mut logical_size) = {
+                let mut window_size = window.size.lock().unwrap();
+                let old_size = *window_size;
+                *window_size = window.new_size.unwrap_or(old_size);
+                (old_size, *window_size)
```

Comment:
It should be `s/old_size/old_logical_size/` or `s/window_size/new_size/` for consistency.